vector3 row0
vector3 row1
vector3 row2
